homeassistant/helpers/event.py

```diff
@@ -630,25 +630,12 @@ def _create_listeners(self) -> None:
         self._setup_entities_listener(self._last_domains, self._last_entities)
 
     @callback
-    def _cancel_domains_listener(self) -> None:
-        if self._domains_listener is None:
+    def _cancel_listener(self, attr_name: str) -> None:
+        attr_ = getattr(self, attr_name)
+        if attr_ is None:
             return
-        self._domains_listener()
-        self._domains_listener = None
-
-    @callback
-    def _cancel_entities_listener(self) -> None:
-        if self._entities_listener is None:
-            return
-        self._entities_listener()
-        self._entities_listener = None
-
-    @callback
-    def _cancel_all_listener(self) -> None:
-        if self._all_listener is None:
-            return
-        self._all_listener()
-        self._all_listener = None
+        attr_()
+        setattr(self, attr_name, None)
```

Comment:
I wonder if we should use a dict to store the listeners to feel less hacky?

```python
unsub = self._listeners.pop(attr_name, None)
if unsub:
    unsub()
```

Reply:
Done